### DISCUSSION TITLE
Is there a way to log all outgoing RESTEasy client requests?

### DISCUSSION THREAD

**Discussion by lucaspottersky** [resteasy/resteasy#4156] [Q&A]
I'm using Wildfly 26. The app uses RESTEasy to reach an API endpoint. I'd like to be able to see these request (like we do with Javascript, with the Chrome Network Panel for example).

**Comment by jamezp** [comment]
Do you mean tracing the calls? Something like the tracing feature https://docs.jboss.org/resteasy/docs/4.7.7.Final/userguide/html/Tracing_Feature.html. Depending on where you want to see it is how you'd configure it. In WildFly 29+ the tracing feature is shipped in WildFly and configurable via the subsystem.

**Comment by lucaspottersky** [comment]
I mean, I'd like to monitor all ResteasyClient calls. I also created this question https://stackoverflow.com/questions/78484181/how-to-log-request-response-using-javax-ws-rs-client-invocation

```
import javax.ws.rs.client.ClientBuilder;
import javax.ws.rs.client.Invocation;
import javax.ws.rs.core.Response;

import org.jboss.resteasy.client.jaxrs.ResteasyClient;
import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
import org.jboss.resteasy.client.jaxrs.ResteasyWebTarget;


public class Main {
    
    public static void main(String[] args) {
        ResteasyClientBuilder rcb = (ResteasyClientBuilder) ClientBuilder.newBuilder();
        ResteasyClient resteasyClient = rcb.build();
        ResteasyWebTarget target = resteasyClient.target("https://api.api-ninjas.com/v1/loremipsum?paragraphs=1");
        Invocation.Builder builder = target.request();

        // Is there a non-programatically (i.e. config) to log all the requests passing through javax.ws.rs lib???
        Response response = builder.get();  
      
    }
}

```

**Reply by jamezp** [reply]
I assume you want to check client requests. You could do this using a `ClientRequestFilter`.

**Comment by lucaspottersky** [comment]
> I assume you want to check client requests

Indeed. 

> You could do this using a ClientRequestFilter

I am not in control of the app, so a configurable way would be the only feasible way for me.


But thank you for your kind reply!

**Reply by jamezp** [reply]
I'm not aware of a configuration option to explicitly track requests. The tracing feature would help, but I guess it depends on what you'd like to see. However, with WildFly 26 the module is not present so you'd need to have control of being able to install it or modify the deployment.

**Reply by liweinan** [reply]
In addition here is the introduction to the usage of the feature within WildFly: https://resteasy.dev/2023/04/19/tracing-feature-in-wildfly/
